flutter/lib/src/screenshot/screenshot_attachment_processor.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:async';
+import 'dart:typed_data';
+import 'dart:ui' as ui show ImageByteFormat;
+
+import 'package:flutter/rendering.dart';
+import 'package:sentry/sentry.dart';
+import '../integrations/native_app_start_integration.dart';
+import 'sentry_screenshot_widget.dart';
+
+// ignore: invalid_use_of_internal_member
+class ScreenshotAttachmentProcessor implements SentryClientAttachmentProcessor {
+  final SchedulerBindingProvider _schedulerBindingProvider;
+  final SentryOptions _options;
+
+  ScreenshotAttachmentProcessor(this._schedulerBindingProvider, this._options);
+
+  /// This is true when the SentryWidget is in the view hierarchy
+  bool get _attachScreenshot => sentryWidgetGlobalKey.currentContext != null;
+
+  @override
+  Future<List<SentryAttachment>> processAttachments(
+      List<SentryAttachment> attachments, SentryEvent event) async {
+    if (event.exceptions == null &&
+        event.throwable == null &&
+        _attachScreenshot) {
+      return attachments;
+    }
+    final schedulerBinding = _schedulerBindingProvider();
+    if (schedulerBinding != null) {
+      final completer = Completer<Uint8List?>();
+      schedulerBinding.addPostFrameCallback((timeStamp) async {
+        final screenshot = await _createScreenshot();
+        completer.complete(screenshot);
+      });
+      final bytes = await completer.future ?? Uint8List.fromList([]);
+      final screenshotAttachment = SentryAttachment.fromScreenshotData(bytes);
+
+      final attachments = <SentryAttachment>[];
+      attachments.addAll(attachments);
+      attachments.add(screenshotAttachment);
```

Comment:
Why do we create a new list and append all + the `screenshotAttachment`?
Can we just append the `screenshotAttachment`?